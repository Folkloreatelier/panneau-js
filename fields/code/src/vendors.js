import Loadable from 'react-loadable';

export const AceEditor = Loadable({
    loader: () => import('react-ace'),
    loading: () => null,
});

export const loadBrace = (language, theme, extensions = []) => (
    import('brace')
        .then(() => Promise.all([
            import(`brace/mode/${language}`),
            import(`brace/theme/${theme}`),
            ...extensions.map(ext => import(`brace/ext/${ext}`)),
        ]))
);
