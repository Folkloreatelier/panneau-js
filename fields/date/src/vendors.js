import Loadable from 'react-loadable';

export const DayPickerRangeController = Loadable({
    loader: () => import('react-dates/lib/components/DayPickerRangeController'),
    loading: () => null,
});

export const DayPickerSingleDateController = Loadable({
    loader: () => import('react-dates/lib/components/DayPickerSingleDateController'),
    loading: () => null,
});
