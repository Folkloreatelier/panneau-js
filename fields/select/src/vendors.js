import Loadable from 'react-loadable';

export const AsyncCreatable = Loadable({
    loader: () => import('react-select/lib/AsyncCreatable'),
    loading: () => null,
});

export const Async = Loadable({
    loader: () => import('react-select/lib/Async'),
    loading: () => null,
});

export const Creatable = Loadable({
    loader: () => import('react-select/lib/Creatable'),
    loading: () => null,
});

export const Select = Loadable({
    loader: () => import('react-select/lib/Select'),
    loading: () => null,
});
