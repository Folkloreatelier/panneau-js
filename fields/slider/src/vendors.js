import Loadable from 'react-loadable';

export const Slider = Loadable({
    loader: () => import('rc-slider/lib/Slider'),
    loading: () => null,
});

export const Range = Loadable({
    loader: () => import('rc-slider/lib/Range'),
    loading: () => null,
});
