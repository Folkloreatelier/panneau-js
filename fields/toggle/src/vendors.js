import Loadable from 'react-loadable';

export default Loadable({
    loader: () => import(/* webpackChunkName: "vendor/rc-switch" */ 'rc-switch'),
    loading: () => null,
});
