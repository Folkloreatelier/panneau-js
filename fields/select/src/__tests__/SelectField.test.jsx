import React from 'react';
import Loadable from 'react-loadable';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';
import { IntlProvider } from 'react-intl';
import { SelectBase, Async as AsyncBase } from 'react-select';
import BaseSelectField from '../SelectField';
import { Select, Async, AsyncCreatable, Creatable } from '../vendors';

const defaultOptions = [
    {
        value: 1,
        label: 'Option 1'
    },
    {
        value: 2,
        label: 'Option 2'
    },
];

const SelectField = props => (
    <IntlProvider locale="en">
        <BaseSelectField {...props} />
    </IntlProvider>
);

test('SelectField match snapshot', async () => {
    await Loadable.preloadAll();

    const selectComponent = renderer.create(<SelectField options={defaultOptions} value={1} />);
    expect(selectComponent.toJSON()).toMatchSnapshot();

    const asyncComponent = renderer.create(<SelectField options={defaultOptions} value={1} async />);
    expect(asyncComponent.toJSON()).toMatchSnapshot();

    const asyncCreatableComponent = renderer.create(<SelectField options={defaultOptions} value={1} async creatable />);
    expect(asyncCreatableComponent.toJSON()).toMatchSnapshot();

    const creatableComponent = renderer.create(<SelectField options={defaultOptions} value={1} creatable />);
    expect(creatableComponent.toJSON()).toMatchSnapshot();
});

test('render Select by default', async () => {
    await Loadable.preloadAll();
    const field = shallow(<SelectField />);
    expect(field.find(Select).length).toEqual(1);
});

test('render Async when async=true', async () => {
    await Loadable.preloadAll();
    const field = shallow(<SelectField async />);
    expect(field.find(Async).length).toEqual(1);
});

test('render AsyncCreatable when async=true and creatable=true', async () => {
    await Loadable.preloadAll();
    const field = shallow(<SelectField async creatable />);
    expect(field.find(AsyncCreatable).length).toEqual(1);
});

test('render Creatable when creatable=true', async () => {
    await Loadable.preloadAll();
    const field = shallow(<SelectField creatable />);
    expect(field.find(Creatable).length).toEqual(1);
});

test('value match option', async () => {
    await Loadable.preloadAll();
    const field = shallow(<SelectField options={defaultOptions} value={1} />);
    expect(field.find(Select).prop('value')).toEqual(defaultOptions[0]);
});

test('value match option when multiple', async () => {
    await Loadable.preloadAll();
    const field = shallow(<SelectField options={defaultOptions} value={[1,2]} multiple />);
    expect(field.find(Select).prop('value')).toEqual(defaultOptions);
});

test('fetch options when async', async () => {
    await Loadable.preloadAll();
    const fetchOptions = () => new Promise(resolve => resolve(defaultOptions));
    const field = mount(<SelectField fetchOptions={fetchOptions} defaultOptions value={[1,2]} async />);
    await new Promise(resolve => setTimeout(resolve, 200));
    expect(field.state('loadedOptions')).toEqual(defaultOptions);
    expect(field.find(AsyncBase).state('defaultOptions')).toEqual(defaultOptions);
});

test('has value when cannotBeEmpty', async () => {
    await Loadable.preloadAll();
    const field = shallow(<SelectField options={defaultOptions} cannotBeEmpty />);
    expect(field.find(Select).prop('value')).toEqual(defaultOptions[0]);
});

test('onChange is called and receive value', async () => {
    const onChange = sinon.spy();
    const field = mount(<SelectField options={defaultOptions} value={1} onChange={onChange} />);
    field.find(Select).prop('onChange')(defaultOptions[0]);
    expect(onChange.calledOnce).toEqual(true);
    expect(onChange.calledWith(1)).toEqual(true);
});
