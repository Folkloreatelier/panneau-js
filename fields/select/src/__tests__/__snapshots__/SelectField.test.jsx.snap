// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField render a <div /> 1`] = `null`;
